File: golang/deepseek.go
package main

import (
	"bytes"
	"encoding/json"
	"fmt"
	"io"
	"log"
	"net/http"
	"os"

	"github.com/joho/godotenv"
)

type RequestPayload struct {
	Model      string    `json:"model"`
	Messages   []Message `json:"messages"`
	Stream     bool      `json:"stream"`
}

type Message struct {
	Role    string `json:"role"`
	Content string `json:"content"`
}

func getAPIKey() string {
	err := godotenv.Load()
	if err != nil {
		log.Fatal("Error loading .env file")
	}
	apiKey := os.Getenv("DEEPSEEK_API_KEY")
	if apiKey == "" {
		log.Fatal("DEEPSEEK_API_KEY not found in environment")
	}
	return apiKey
}

func makeRequest(prompt string) (string, error) {
	apiKey := getAPIKey()
	baseURL := "https://api.deepseek.com/chat/completions"

	payload := RequestPayload{
		Model: "deepseek-chat",
		Messages: []Message{
			{"system", "You are a helpful assistant."},
			{"user", prompt},
		},
		Stream: false,
	}

	jsonData, err := json.Marshal(payload)
	if err != nil {
		return "", fmt.Errorf("error marshalling JSON: %v", err)
	}

	req, err := http.NewRequest("POST", baseURL, bytes.NewBuffer(jsonData))
	if err != nil {
		return "", fmt.Errorf("error creating request: %v", err)
	}
	req.Header.Set("Content-Type", "application/json")
	req.Header.Set("Authorization", "Bearer "+apiKey)

	client := &http.Client{}
	resp, err := client.Do(req)
	if err != nil {
		return "", fmt.Errorf("error sending request: %v", err)
	}
	defer resp.Body.Close()

	body, err := io.ReadAll(resp.Body)
	if err != nil {
		return "", fmt.Errorf("error reading response: %v", err)
	}

	if resp.StatusCode != http.StatusOK {
		return "", fmt.Errorf("API error %d: %s", resp.StatusCode, body)
	}

	return string(body), nil
}

func main() {
	response, err := makeRequest("Hello, how are you?")
	if err != nil {
		log.Fatalf("Request failed: %v", err)
	}
	fmt.Println(response)
	fmt.Println("\n🐬")
}

File: ts/deepseek.ts
const axios = require("axios");
const dotenv = require("dotenv");

dotenv.config();

const DEEPSEEK_API_KEY = process.env.DEEPSEEK_API_KEY;

if (!DEEPSEEK_API_KEY) {
    console.error("DEEPSEEK_API_KEY not found in environment variables.");
    process.exit(1);
}

async function makeRequest(
    prompt: string,
    model: string = "deepseek-chat",
    systemRole: string = "You are a helpful assistant.",
    stream: boolean = false
) {
    const baseURL = "https://api.deepseek.com/chat/completions";

    const headers = {
        "Content-Type": "application/json",
        Authorization: `Bearer ${DEEPSEEK_API_KEY}`,
    };

    const payload = {
        model,
        messages: [
            { role: "system", content: systemRole },
            { role: "user", content: prompt },
        ],
        stream,
    };

    try {
        const response = await axios.post(baseURL, payload, { headers });
        console.log("Response:", response.data);
    } catch (error: any) {
        console.error("Request failed:", error.response ? error.response.data : error.message);
    }
}

makeRequest("Hello, how are you?");
